src/page-modules/assistant/server/journey-planner/index.ts

```diff
@@ -157,12 +167,86 @@ export function createJourneyApi(
 
       const from = inputToLocation(input, 'from');
       const to = inputToLocation(input, 'to');
-
       const when =
         input.searchTime.mode !== 'now'
           ? new Date(input.searchTime.dateTime)
           : new Date();
 
+      if (input.via) {
```

Comment:
I think all of this could maybe be seperated into different functions? This function body is quite big now. Hard to separate